src/current.rs

```diff
@@ -27,19 +25,19 @@ pub fn mcu() -> Option<Mcu> {
 /// * `Some("attiny85")`
 /// * `None`
 pub fn mcu_name() -> Option<String> {
-    if is_targeting_avr() {
-        // The avr-rust compiler fork has a special configuration option
-        // that exposes the target microcontroller name in lowercase.
-        match env::var("CARGO_CFG_TARGET_CPU") {
-            Ok(target_cpu) => Some(target_cpu),
-            Err(..) => panic!("cannot retrieve mcu name, use a version of the avr-rust compiler that exposes the 'target_cpu' configuration option"),
-        }
-    } else {
-        None
-    }
-}
+    // TARGET environment variable should contain the value that was specified
+    // by --target X(C)argo option. Normally it's the name of .json file
+    // containing custom target specification, e. g. atmega88pa.json
+    // So in order to work, the name of *.json file should be the same
+    // as the name of your MCU
+    let target = env::var("TARGET").expect(
```

Comment:
TIL. this is really nice, better than adding AVR specific hacks to the compiler.